### PR TITLE
Add rebase-to-origin.sh script

### PR DESCRIPTION
**Pull request by mfojtik**
Tired of manual rebase to origin? Try this handy script :-)

This script assumes that the "origin" is cloned into $GOPATH/src/github.com/openshift/origin.
This script will create a new branch, so make sure the "origin" HEAD is the latest master.
After you invoke this script, the branch and the commit will be created with proper commit message, ready to push and create a PR from it.

### CODE REVIEW

**Comment by mfojtik** [comment]
@soltysh @bparees FYI

File: hack/rebase-to-origin.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  # Remove all explicitely excluded packages
+  for pkg in "${exclude_pkgs[@]}"; do
+    rm -rf "${origin_s2i_godep_dir}/${pkg}"
+  done
+
+  # Bump the origin Godeps.json file
+  sed -i '' "s/${s2i_godeps_ref}/${s2i_ref}/g" ${origin_dir}/Godeps/Godeps.json
+
+  # Make a commit with proper message
+  git add Godeps && git commit -m "bump(github.com/openshift/source-to-image) ${s2i_ref}"
```
Comment (soltysh) [inline hack/rebase-to-origin.sh]: The form is: bump(path): commit id  <- you're missing the colon after closing bracket.
See https://github.com/openshift/origin/blob/master/HACKING.md#updating-other-godeps-from-upstream

**Comment by soltysh** [comment]
One nit, otherwise LGTM.

**Comment by mfojtik** [comment]
@soltysh corrected, thanks! [merge]

**Comment by bparees** [comment]
so we've given up on using the "godeps" tool entirely eh?  maybe that should tell us something instead of writing bash scripts to work around it :(

**Comment by mfojtik** [comment]
@bparees well godeps save/restore on origin does not work anyway as we ship the extra patches to other projects...

**Comment by soltysh** [comment]
There's a WIP for replacing godeps with something different, to be chosen what exactly.

**Comment by rhcarvalho** [comment]
if this is to exist, shouldn't it live in origin instead of here?

**Comment by mfojtik** [comment]
@rhcarvalho yes it could and I will remove this file as soon as we have sane way to do rebase in origin for S2I :-) Without that I'm keeping this file there so whoever needs a rebase to origin don't need to type the commands in script.

**Pull request merged**